liblangutil/Scanner.cpp

```diff
@@ -271,12 +273,67 @@ bool Scanner::skipWhitespaceExceptUnicodeLinebreak()
 	return sourcePos() != startPosition;
 }
 
+struct Scanner::UnicodeDirectionOverrideProcessor
+{
+	Scanner& scanner;
+	int rtlOverrideDepth = 0;
+
+	pair<bool, ScannerError> operator()()
+	{
+		// U+202D (LRO - Left-to-Right Override)
+		// U+202E (RLO - Right-to-Left Override)
+		if (
+			scanner.tryScanByteSequence("\xE2\x80\xAD") ||
+			scanner.tryScanByteSequence("\xE2\x80\xAE")
+		)
+		{
+			rtlOverrideDepth++;
+			return pair{true, ScannerError::NoError};
```

Comment:
I think this should work without `pair` too:
```suggestion
			return {true, ScannerError::NoError};
```

Reply:
It does, but I had problems on MSVC recently (not this PR) where MSVC could not deduce the type, so I had to fully qualify it. I'm fine though if it works, it works. ;)

Reply:
Right, I've had that in a different situation too.

I think it's redundant but I also don't have anything against it if it's needed to satisfy the compiler.